backend/src/test/java/com/bakdata/conquery/models/SerializationTests.java

```diff
@@ -150,6 +155,119 @@ public void group() throws IOException, JSONException {
 				.test(group);
 	}
 
+
+	@Test
+	public void bucketCompoundDateRange() throws JSONException, IOException {
+		Dataset dataset = new Dataset();
+		dataset.setName("datasetName");
+
+		Table table = new Table();
+
+		Column startCol = new Column();
+		startCol.setName("startCol");
+		startCol.setType(MajorTypeId.DATE);
+		startCol.setTable(table);
+
+		Column endCol = new Column();
+		endCol.setLabel("endLabel");
+		endCol.setName("endCol");
+		endCol.setType(MajorTypeId.DATE);
+		endCol.setTable(table);
+
+
+		Column compoundCol = new Column();
+		compoundCol.setName("compoundCol");
+		compoundCol.setType(MajorTypeId.DATE_RANGE);
+		compoundCol.setTable(table);
+
+		table.setColumns(new Column[]{startCol, endCol, compoundCol});
+		table.setDataset(dataset);
+		table.setName("tableName");
+
+
+		Import imp = new Import(table);
+		imp.setName("importTest");
+
+
+		DateRangeTypeCompound
+				compoundStore =
+				new DateRangeTypeCompound(startCol.getName(), endCol.getName(), new BitSetStore(BitSet.valueOf(new byte[]{0b1000}), new BitSet())); //0b1000 is a binary representation of 8 so that the 4th is set to make sure that BitSet length is 4.
```

Comment:
Warum ist es hier wichtig, dass values eine gesetzte länge hat, aber nullBits nicht?

Reply:
Es gibt doch auch BitSetStore.create

Reply:
die BitSetStore.create erstellt ein Store von BitSet mit reserviertem Speicherplatz, der aber noch nicht belegt ist. Deswegen kriegt man 0 wenn man getLines auf dem Store aufruft. Von daher setze beim Erstellen die Bits.

Reply:
die nullBits müssen nicht gesetzt werden da ich schon meine gewollte Länge mit values erreiche. Ich hätte das auch mit nullBits nutzen können. Letztendlich ist es für diesen Test egal welcher von den beiden Nutzung findet

Reply:
> die BitSetStore.create erstellt ein Store von BitSet mit reserviertem Speicherplatz, der aber noch nicht belegt ist. 

Das ist so wie ich die API gestalten wollte dann aber ein Bug. Kannst du den beheben? Vielleicht muss für den BitsetStore die länge manuell getracked werden. @thoniTUB ist ja auch aufgefallen, dass wir übersehen hatten, dass not-null dangling-false nicht in der länge betrachtet werden

Reply:
> die nullBits müssen nicht gesetzt werden da ich schon meine gewollte Länge mit values erreiche. Ich hätte das auch mit nullBits nutzen können. Letztendlich ist es für diesen Test egal welcher von den beiden Nutzung findet

Das stimmt aber auch!

Reply:
https://github.com/bakdata/conquery/pull/2114#discussion_r769576737

Reply:
Wie ich erwähnt habe, sehe ich nicht wie man irgend wie aus den bisherigen Bitstores die Länge herleiten kann, also bitte als extra Parameter.